build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/precommit/JarHellPrecommitPlugin.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle.internal.precommit;
+
+import org.elasticsearch.gradle.internal.conventions.precommit.PrecommitPlugin;
+import org.elasticsearch.gradle.internal.info.BuildParams;
+import org.elasticsearch.gradle.jarhell.JarHellPlugin;
+import org.elasticsearch.gradle.jarhell.JarHellTask;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.tasks.TaskProvider;
+
+public class JarHellPrecommitPlugin extends PrecommitPlugin {
+    @Override
+    public TaskProvider<? extends Task> createTask(Project project) {
+        project.getPluginManager().apply(JarHellPlugin.class);
+
+        if (BuildParams.isInternal() && project.getPath().equals(":libs:elasticsearch-core") == false) {
```

Comment:
We shouldn't need to reference `isInternal` here, right or is just to be able to test this?

Reply:
exactly. this is needed to keep the tests working without too much hassle now. We'll remove this `isInternal()` completely at one point. Also we'll remove default dependencies declarations in plugins too. Ideally we'll provide a bom that can be used in our build. that way we'll be able to use different versions with the same plugin.